Add a caution regarding the `exec` probe mechanism (#41286)
kubernetes/website: content/en/docs/concepts/workloads/pods/pod-lifecycle.md

  the port is open. If the remote system (the container) closes
  the connection immediately after it opens, this counts as healthy.

{{< caution >}} Unlike the other mechanisms, `exec` probe's implementation involves the creation/forking of multiple processes each time when executed.
As a result, in case of the clusters having higher pod densities, lower intervals of `initialDelaySeconds`, `periodSeconds`, configuring any probe with exec mechanism might introduce an overhead on the cpu usage of the node.

> In such scenarios, the usage of other probe mechanisms is highly recommended to avoid the overhead.{{< /caution >}}

Reviewer: s/mechanisms/types perhaps

> In such scenarios, the usage of other probe mechanisms is highly recommended to avoid the overhead.{{< /caution >}}

Reviewer 2: ```suggestion
In such scenarios, consider using alternative probe mechanisms is recommended to avoid the overhead.{{< /caution >}}
```

As a project, Kubernetes does not “highly recommend” avoiding exec probes. Cluster operators can make their own decisions.

Author: Yeah, Updated. Thanks !
